server/src/main/java/org/cloudfoundry/identity/uaa/provider/IdentityProviderEndpoints.java

```diff
@@ -452,18 +427,15 @@ protected void redactSensitiveData(IdentityProvider provider) {
         }
         switch (provider.getType()) {
             case LDAP: {
-                if (provider.getConfig() instanceof LdapIdentityProviderDefinition) {
-                    logger.debug("Removing bind password from LDAP provider id:"+provider.getId());
-                    LdapIdentityProviderDefinition definition = (LdapIdentityProviderDefinition) provider.getConfig();
+                if (provider.getConfig() instanceof LdapIdentityProviderDefinition definition) {
+                    logger.debug("Removing bind password from LDAP provider id: {}", provider.getId());
                     definition.setBindPassword(null);
                 }
                 break;
             }
-            case OAUTH20 :
-            case OIDC10 : {
-                if (provider.getConfig() instanceof AbstractExternalOAuthIdentityProviderDefinition) {
-                    logger.debug("Removing relying secret from OAuth/OIDC provider id:"+provider.getId());
-                    AbstractExternalOAuthIdentityProviderDefinition definition = (AbstractExternalOAuthIdentityProviderDefinition) provider.getConfig();
+            case OAUTH20, OIDC10 : {
+                if (provider.getConfig() instanceof AbstractExternalOAuthIdentityProviderDefinition<?> definition) {
+                    logger.debug("Removing relying secret from OAuth/OIDC provider id: {}", provider.getId());
```

Comment:
Why not checking for `logger.isDebugEnabled()` here?

Reply:
Sonar does not want it in case of normal getters. In other cases there is a real change of input and thus this is wanted to be invoked only conditionally -> rule java:S2629